### ISSUE TITLE
MSSM_2L_limits test fails on openSUSY Tumbleweed

### ISSUE THREAD

**Issue by wkotlarski** [FlexibleSUSY/FlexibleSUSY#150]
The error message is
```
test/test_MSSM_2L_limits.cpp(334): error: in "MSSM_dMh_at_at_st_0_mst1_eq_mst2": difference{0.143097} between dMh_ps(0,1){2.3869875055284897} and dMh_fs_exact(0,1){2.7285590464895115} exceeds 7%
test/test_MSSM_2L_limits.cpp(335): error: in "MSSM_dMh_at_at_st_0_mst1_eq_mst2": difference{0.143097} between dMh_ps(1,0){2.3869875055284897} and dMh_fs_exact(1,0){2.7285590464895115} exceeds 7%
test/test_MSSM_2L_limits.cpp(336): error: in "MSSM_dMh_at_at_st_0_mst1_eq_mst2": difference{6.19315e-05} between dMh_ps(1,1){-1115.0422434243928} and dMh_fs_exact(1,1){-1115.1112997172997} exceeds 0.004%
```
Does anyone know what can be the cause of this? Or how to debug it? It works for @uukhas on Ubuntu 18.04 and on openSUSY Leap. Maybe it's because Tumbleweed has newest versions of packages?

**Comment by Expander** [comment]
Can you please print the verbose output by running
~~~
test/test_MSSM_2L_limits.x --log_level=test_suite
~~~
?

I think this test is not crucial. It checks numerically some limits of the exact 2-loop routines from Pietro Slavich against the proper implementation of these limits in FS. Since the test works in a divergent region, it is not completely surprising that it fails.

However, the precise reason for the failure may point to some numerical instability, which may have been introduced in Tumbleweed . Maybe due to some compiler flags, like `-O3` or `-fast-math` or similar?

**Comment by wkotlarski** [comment]
Here's the output you asked for [test_MSSM_2L_limits.log](https://github.com/FlexibleSUSY/FlexibleSUSY/files/4255844/test_MSSM_2L_limits.log). I don't change compiler flags (so no `-O3` or `-fast-math`).

**Comment by Expander** [comment]
I see the following:
~~~
Pietro Slavich:
 2.21147    2.38699
 2.38699   -1115.04
Limit st -> 0 and mst1 -> mst2:
 0.503647   2.72856
 2.72856   -1115.11
~~~
On my machine I get:
~~~
Pietro Slavich                 :
0.370989  2.75508
 2.75508 -1115.12
Limit st -> 0 and mst1 -> mst2 :
-0.549568    2.9392
   2.9392  -1115.15
~~~
From looking into that is tested, I see that it is the function `self_energy_higgs_2loop_at_at_mssm_with_tadpoles`.  This function directly calls the Slavich routines `ddshiggs_`. Inside `ddshiggs_` there are checks to not run into the divergent limits. It seems that these checks don't quite work on Tumbleweed.

So, I now believe the issue has to do with the used Fortran compiler. Which one do you use?

**Comment by Expander** [comment]
Or, it may be that the Fortran code is optimized in such a way that the instability is more visible.

**Comment by wkotlarski** [comment]
I use gfortran version 9.2.1

**Comment by Expander** [comment]
OK.

I think this test should be fixed by decreasing the test precision, because it is obviously sensitive to numerical cancellations, which again are sensitive to the chosen compilers/flags/...

**Comment by wkotlarski** [comment]
fixed in 14d096cbd1e7729fb9d5e9beb5cfe7b43465d5ba
